inc/clr/prof/corprof.h

```diff
@@ -1,4 +1,6 @@
 
```

Comment:
This file is pulled from the clr repo, we may want to just supress these warnings. The next time we updated it, we'll have to go through this again.

Reply:
I had to fix the SAL annotations in this file because otherwise preFAST was complaining the overridden methods in the derived classes had mismatched annotations. I had three options: disable SAL annotation checks, remove the annotations in the derived classes or add them in corprof.h
I did find some wrong SAL annotations, so I guess disabling the warning wouldn't be the best thing.

Reply:
SAL warnings should just be suppressed for this file. I think we do something similar in the aerobee repository. Maybe we can have a chat about how to fix this.

Reply:
In the aerobee repository, we added a separate file called corprofiler.h:

```
#pragma once
#pragma warning(push) // corprof.h from the coreclr does not include SAL anotations
#pragma warning(disable : 6313 25057 25155 28718 28722 28723 28729 28740)
#include "corprof.h"
#pragma warning(pop)
```

We never reference corprof.h directly, instead we include "corprofiler.h" I would suggest doing the same here.


---
In reply to: [520710879](https://github.com/microsoft/CLRInstrumentationEngine/pull/325#discussion_r520710879) [](ancestors = 520710879)